File: web/src/api/yunguan/resource/resDisk.js
import request from '@/utils/request'

// 添加元素
export function addDisk(modelData) {
    return request({
        url: 'res/disk/addDisk',
        method: 'post',
        data: modelData
    })
}

// 修改元素
export function updateDisk(params) {
    return request({
        url: 'res/disk/updateDisk',
        method: 'put',
        data: params
    })
}

// 获取节点元素列表
export function getDiskList(params) {
    return request({
        url: 'res/disk/queryDisk',
        method: "post",
        data:params
    })
}

// 删除元素
export function deleteDisk(params) {
    return request({
        url: 'res/disk/deleteDisk',
        method: "delete",
        data: {
            id: params
        }
    })
}

File: server/router/yunguan/res/resDisk.go
package res

import (
	"github.com/gin-gonic/gin"
	v1 "ygpt/server/api/v1"
)

type ResDiskRouter struct {
}

func (s *ResDiskRouter) InitResDiskRouter(Router *gin.RouterGroup) {
	dataRouter := Router.Group("res/disk")
	diskApi := v1.ApiGroupApp.YunGuanApiGroup.ResApi.DiskApi

	{
		dataRouter.POST("addDisk", diskApi.AddDisk)         // 添加数据
		dataRouter.PUT("updateDisk", diskApi.UpdateDisk)    // 修改数据
		dataRouter.DELETE("deleteDisk", diskApi.DeleteDisk) // 删除数据
		dataRouter.POST("queryDisk", diskApi.QueryDisk)     // 获取数据列表
	}
}
